modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/SCIM2RestClient.java

```diff
@@ -0,0 +1,232 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.identity.integration.test.restclients;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.testng.Assert;
+import org.wso2.carbon.automation.engine.context.beans.Tenant;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.identity.integration.test.rest.api.user.common.model.PatchOperationRequestObject;
+import org.wso2.identity.integration.test.rest.api.user.common.model.RoleRequestObject;
+import org.wso2.identity.integration.test.rest.api.user.common.model.RoleSearchRequestObject;
+import org.wso2.identity.integration.test.rest.api.user.common.model.UserObject;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class SCIM2RestClient extends RestBaseClient {
+
+    private static final String SCIM2_USERS_ENDPOINT = "scim2/Users";
+    private static final String SCIM2_ROLES_ENDPOINT = "scim2/Roles";
+    private static final String SCIM2_ROLE_SEARCH_PATH = "/.search";
+    private static final String SCIM_JSON_CONTENT_TYPE = "application/scim+json";
+    private static final String ROLE_SEARCH_SCHEMA = "urn:ietf:params:scim:api:messages:2.0:SearchRequest";
+    private static final String DISPLAY_NAME_ATTRIBUTE = "displayName";
+    private static final String EQ_OP = "eq";
+    private final String serverUrl;
+    private final String tenantDomain;
+    private final String username;
+    private final String password;
+
+    public SCIM2RestClient(String serverUrl, Tenant tenantInfo){
+        this.serverUrl = serverUrl;
+        this.tenantDomain = tenantInfo.getContextUser().getUserDomain();
+        this.username = tenantInfo.getContextUser().getUserName();
+        this.password = tenantInfo.getContextUser().getPassword();
+    }
+
+    /**
+     * Create a user
+     *
+     * @param userInfo object with user creation details.
+     * @return Id of the created user.
+     */
+    public String createUser(UserObject userInfo) throws Exception {
+        String jsonRequest = toJSONString(userInfo);
```

Comment:
Shall we add the missing new line?

Reply:
Fixed with https://github.com/wso2/product-is/pull/20096